src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2730,109 +2804,85 @@ private sealed class MemberInitializationUpdates(INamedTypeSymbol oldType)
-                                    {
-                                        if (!TryAddParameterInsertOrDeleteEdits(
-                                            semanticEdits,
-                                            newParameter,
-                                            newModel,
-                                            oldModel,
-                                            capabilities,
-                                            syntaxMap,
-                                            editScript,
-                                            processedSymbols,
-                                            cancellationToken,
-                                            out var notSupportedByRuntime))
+                                        var containingTypeKey = symbolCache.GetKey(newContainingType, cancellationToken);
+                                        oldContainingType = containingTypeKey.Resolve(oldCompilation, ignoreAssemblyKey: true, cancellationToken).Symbol as INamedTypeSymbol;
+
+                                        // Check rude edits for each member even if it is inserted into a new type.
+                                        if (!hasAssociatedSymbolInsert && IsMember(newSymbol))
                                         {
-                                            diagnostics.Add(new RudeEditDiagnostic(
-                                                notSupportedByRuntime ? RudeEditKind.InsertNotSupportedByRuntime : RudeEditKind.Insert,
-                                                GetDiagnosticSpan(newDeclaration, EditKind.Insert),
-                                                newDeclaration,
-                                                arguments: new[] { GetDisplayName(newDeclaration, EditKind.Insert) }));
+                                            ReportInsertedMemberSymbolRudeEdits(diagnostics, newSymbol, newDeclaration, insertingIntoExistingContainingType: oldContainingType != null);
                                         }
 
-                                        continue;
-                                    }
-
-                                    if (newContainingType != null && !IsGlobalMain(newSymbol))
-                                    {
-                                        // The edit actually adds a new symbol into an existing or a new type.
+                                        if (oldContainingType == null)
+                                        {
+                                            // If a type parameter is inserted into the parameter list of a type declaration, the symbol key won't be resolved (because the arities do not match).
+                                            if (!hasAssociatedSymbolInsert && newSymbol is ITypeParameterSymbol)
+                                            {
+                                                diagnosticContext.Report(RudeEditKind.Insert, cancellationToken);
```

Comment:
Can this be treated as a new type definition?

Reply:
Maybe. It'd be a bit complicated since we would need to check if any of the members of the type are active. We wouldn't be able to make the update in such case. Lambdas and state machines defined in the type might also be problematic.